src/mudlet.cpp

```diff
@@ -4122,6 +4122,553 @@ void mudlet::playSound(const QString& s, int soundVolume)
+            return fileNameList;
+        } else {
+            if (!soundType.isEmpty()) {
+                QString soundTypePath = QStringLiteral("%1/%2").arg(mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName()), soundType);
+                QDir soundTypeDir = mudlet::self()->getMSPDir(soundTypePath);
+
+                if (soundTypeDir.isEmpty()) {
+                    qWarning() << QStringLiteral(
+                        "mudlet::getMSPFileNameList() WARNING - Attempt made to create a directory failed: %1")
+                            .arg(mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName()), soundType);
+                    return fileNameList;
+                }
+
+                fileNameList = mudlet::self()->parseMSPFileNameList(true, soundFileName, soundType, soundTypeDir);
+            }
+
+            // Enter this block if no soundType was specified.  Also, per the specification, if soundType was specified above, but we did not
+            // find anything in a soundType directory, fall back and search for the soundFileName in the root "sound" directory.
+            if (fileNameList.isEmpty()) {
+                fileNameList = mudlet::self()->parseMSPFileNameList(true, soundFileName, nullptr, soundDir);
+            }
+        }
+    }
+
+    return fileNameList;
+}
+
+QUrl mudlet::getMSPFileUrl(QString& soundFileName, const QString& soundType)
+{
+    QUrl fileUrl;
+    QPointer<Host> pHost = getActiveHost();
+
+    if (pHost && !pHost->getMSPSoundLocation().isEmpty()) {
+        bool endsWithSlash = pHost->getMSPSoundLocation().endsWith('/');
+
+        if (!soundType.isEmpty()) {
+            if (!endsWithSlash) {
+                fileUrl = QUrl::fromUserInput(QStringLiteral("%1/%2/%3").arg(pHost->getMSPSoundLocation(), soundType, soundFileName));
+            } else {
+                fileUrl = QUrl::fromUserInput(QStringLiteral("%1/%2%3").arg(pHost->getMSPSoundLocation(), soundType, soundFileName));
+            }
+        } else {
+            if (!endsWithSlash) {
+                fileUrl = QUrl::fromUserInput(QStringLiteral("%1/%2").arg(pHost->getMSPSoundLocation(), soundFileName));
+            } else {
+                fileUrl = QUrl::fromUserInput(QStringLiteral("%1%2").arg(pHost->getMSPSoundLocation(), soundFileName));
+            }
+        }
+    }
+
+    return fileUrl;
+}
+
+bool mudlet::downloadMSPFile(QString& soundFileName, const QString& soundType, QString& absolutePathFileName)
+{
+    bool wasFileDownloaded = false;
+    QPointer<Host> pHost = getActiveHost();
+
+    if (pHost) {
+        QDir dir;
+        QString cacheDir = QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
+        if (!dir.mkpath(cacheDir)) {
+            qWarning() << "mudlet::playMSPSound() WARNING - Couldn't create cache directory for sound file(s): " << cacheDir;
+            return wasFileDownloaded;
+        }
+
+        auto manager = new QNetworkAccessManager(this);
```

Comment:
Not yet *another* `QNetworkAccessManage` instance - according to the Qt documentation https://doc.qt.io/qt-5/qnetworkaccessmanager.html#details we only need one instance for the WHOLE application - yet we create multiple ones for each profile and here is *another* one... *sigh* :slightly_frowning_face:

Reply:
I saw some commentary on this being a larger problem as well.  If we don't mind, I'd like to leave this and allow a core lib person to resolve this for the whole application once this PR is closed.

Reply:
Yeah - we can erect a ***[SEP](https://en.wikipedia.org/wiki/Somebody_else's_problem#Douglas_Adams)*** barrier around it for the moment... :grinning:

Reply:
Lol.  I will consider tackling this in the next version if no one else gets to it.